CRM/SMS/Provider.php

```diff
@@ -233,38 +244,41 @@ public function processInbound($from, $body, $to = NULL, $trackID = NULL) {
         ),
       );
       $fromContact = CRM_Contact_BAO_Contact::create($contactparams, FALSE, TRUE, FALSE);
-      $fromContactID = $fromContact->id;
+      $message->fromContactID = $fromContact->id;
     }
 
-    if ($to) {
-      $to = CRM_Utils_Type::escape($to, 'String');
-      $toContactID = CRM_Core_DAO::singleValueQuery('SELECT contact_id FROM civicrm_phone JOIN civicrm_contact ON civicrm_contact.id = civicrm_phone.contact_id WHERE !civicrm_contact.is_deleted AND phone LIKE "%' . $to . '"');
-    }
-    else {
-      $toContactID = $fromContactID;
+    if (!$message->toContactID) {
+      // find recipient if $toContactID not set by hook
+      if ($message->to) {
+        $message->toContactID = CRM_Core_DAO::singleValueQuery("SELECT contact_id FROM civicrm_phone JOIN civicrm_contact ON civicrm_contact.id = civicrm_phone.contact_id WHERE !civicrm_contact.is_deleted AND phone LIKE %1", array(
+          1 => array('%' . $message->to, 'String')));
+      }
+      else {
+        $message->toContactID = $message->fromContactID;
+      }
     }
 
-    if ($fromContactID) {
+    if ($message->fromContactID) {
       $actStatusIDs = array_flip(CRM_Core_OptionGroup::values('activity_status'));
-      $activityTypeID = CRM_Core_OptionGroup::getValue('activity_type', 'Inbound SMS', 'name');
```

Comment:
@eileenmcnaughton stumbled upon another one of these (there were no tests before so Jenkins wasn't getting annoyed)